base/sparse/linalg.jl

```diff
@@ -532,10 +531,13 @@ function normestinv{T}(A::SparseMatrixCSC{T}, t::Integer = 2)
     end
     ind = Array(Int64, n)
     ind_hist = Array(Int64, maxiter * t)
-    S = zeros(T, n, t)
+
+    Ti = T <: Real ? Float64 : Complex128
```

Comment:
`est = zero(real(eltype(Y)))` will still give a real value, so we will not be seeing a complex value for the norm.

As for the double precision, maybe
```julia
Ti = T <: Real ? typeof(float(zero(T))) : typeof(float(complex(zero(T))))
```
would do the trick?